### ISSUE TITLE
Old folder shown while loading new folder

### ISSUE THREAD

**Issue by scottkosty** [neomutt/neomutt#3573]
## Expected Behaviour

While changing folders, I'm used to the folder selection screen showing rather than the previous folder.

## Actual Behaviour

While the new folder is loading, the old one is shown, instead of the folder list.

## Steps to Reproduce

Using IMAP (I can't reproduce on Maildir because folder loading is so fast with Maildir):

1. Press <kbd>c</kbd>.
2. Press <kbd>Tab</kbd>.
3. Select a folder that takes a non-trivial time to load and press <kbd>Enter</kbd>.

## How often does this happen?

- Always (on IMAP)

## When did it start to happen?

Bisect leads to eee52ba9a074c2e20bd15ccd96194f51dc59688a.

## NeoMutt Version

```
NeoMutt 20220429-245-30e895
Copyright (C) 1996-2022 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.15.5-051505-generic (x86_64)
ncurses: ncurses 6.3.20211021 (compiled with 6.3.20211021)
libidn: 1.38 (compiled with 1.38)
OpenSSL: OpenSSL 3.0.2 15 Mar 2022
libnotmuch: 5.6.0
storage: tokyocabinet

Configure options: --ssl --sasl --enable-notmuch --enable-debug --tokyocabinet --prefix=/usr/local

Compilation CFLAGS: -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/local/include -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo -gnutls -gpgme 
  -gsasl -gss +hcache -homespool +idn +inotify -locales_hack -lua -mixmaster 
  +nls +notmuch +openssl +pgp +regex +sasl +smime -sqlite +sun_attachment 

MAILPATH="/var/mail"
PKGDATADIR="/usr/local/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/usr/local/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Is your email local (maildir) or remote (IMAP)?

IMAP.

**Comment by gahr** [comment]
I'm not sure I'd consider this a bug. It is a change in behaviour, but is it really causing issues? IIRC - @flatcap can confirm - this is needed because of a larger refactoring project.

**Comment by scottkosty** [comment]
Thanks for taking a look! I changed the tags to a feature request.

I'm sure I can get used to the new behavior with time. The main workflow that I need to adapt my mind to is the following: I try keep my Inbox at zero. So when I log in, process my inbox, then switch folders, I immediately think "oh there's no new email in the new folder" (because my inbox is shown when switching folders). I think another reason why it will take time to adapt is that the behavior is now different when using Maildir than when using IMAP. With Maildir, the "change folder" screen is shown and then the new folder screen is shown. With IMAP, the change folder screen is shown, then the old folder, then the new folder. So I need to teach my mind to be context-dependent :)

**Comment by gahr** [comment]
> With Maildir, the "change folder" screen is shown and then the new folder screen is shown. With IMAP, the change folder screen is shown, then the old folder, then the new folder.

Is that perhaps because maildir is so fast you don't even see the old index? :)

**Comment by scottkosty** [comment]
> Is that perhaps because maildir is so fast you don't even see the old index? :)

Yes. Maybe I should plug in one of my old 5400 RPM HDD's I have laying around to see if that makes it so the behavior is consistent :)

In any case, when testing 'main' I expect to run into these issues, and when updating versions (sounds like this new behavior will be kept) I need to adapt. And I'm happy to do so! Thanks for all of your work!

**Comment by gahr** [comment]
Btw, I gave this another try and I can't reproduce it. I see a blank index while "Fetching message headers...".

**Comment by scottkosty** [comment]
> Btw, I gave this another try and I can't reproduce it. I see a blank index while "Fetching message headers...".

Thank you very much for trying!
